t/job-manager/plugins/resource-update-expiration.c

```diff
@@ -0,0 +1,85 @@
+/************************************************************\
+ * Copyright 2023 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* jobspec-update-job-list.c - test jobspec-update event in job-list
+ * module
+ */
+
+#include <sys/time.h>
+
+#include <jansson.h>
+
+#include <flux/core.h>
+#include <flux/jobtap.h>
+
+#include "ccan/str/str.h"
+#include "src/common/libutil/errprintf.h"
+
+static int run_cb (flux_plugin_t *p,
+                   const char *topic,
+                   flux_plugin_arg_t *args,
+                   void *data)
+{
+    struct timeval tv;
+    double newexp;
+    flux_jobid_t id;
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:I}",
+                                "id", &id) < 0) {
+        flux_jobtap_raise_exception (p,
+                                     FLUX_JOBTAP_CURRENT_JOB,
+                                     "resource-update", 0,
+                                     "unpack failure");
+        return -1;
+    }
+
+    /* tests expecting an expiration greater than original.  Instead
+     * of getting R and calculating future time, just add 72 hours to
+     * the current time.
+     */
+
+    if (gettimeofday (&tv, NULL) < 0) {
+        flux_jobtap_raise_exception (p,
+                                     FLUX_JOBTAP_CURRENT_JOB,
+                                     "resource-update", 0,
+                                     "gettimeofday");
+        return -1;
+    }
+
+    newexp = tv.tv_sec + (72 * 60 * 60);
+    if (flux_jobtap_event_post_pack (p,
+                                     id,
+                                     "resource-update",
+                                     "{s:f}",
+                                     "execution.expiration", newexp) < 0) {
```

Comment:
As noted elsewhere, RFC 21 specifies `expiration` as the only allowed `resource-update` context key.